functions/Measure-DbaDiskSpaceRequirement.ps1

```diff
@@ -0,0 +1,355 @@
+                } catch {
+                    Write-Warning "Can't connect to $ComputerName. CacheMP[$ComputerName] = ?"
+                    $CacheMP.Add($ComputerName, '?') # This way, I won't be asking again for this computer.
+                }
+            }
+            if($CacheMP[$ComputerName] -eq '?') {
+                return '?'
+            }
+            foreach($M in $CacheMP[$ComputerName]) {
+                if($Path -like "$($M.Name)*") {
+                    return $M.Name
+                }
+            }
+            Write-Warning "Path $Path can't be found in any MountPoints of $ComputerName"
+        }
+
+        function Get-MountPointFromDefaultPath {
+            # Extract MountPoint from DefaultPath. Usefull when database or file does not exist on destination.
+            [CmdletBinding()]
+            Param(
+                [Parameter(Mandatory)]
+                [ValidateSet('Log','Data')]
+                $DefaultPathType,
+                [Parameter(Mandatory)]
+                $SqlInstance,
+                [PSCredential]$SqlCredential,
+                $ComputerName, # Could probably use the computer defined in SqlInstance but info was already available from the caller
+                [PSCredential]$Credential
+            )
+            if(!$CacheDP[$SqlInstance]) {
+                try {
+                    $CacheDP.Add($SqlInstance, (Get-DbaDefaultPath -SqlInstance $SqlInstance -SqlCredential $SqlCredential -EnableException))
+                    Write-Verbose "CacheDP[$SqlInstance] is in cache"
+                } catch {
+                    Write-Warning "Can't connect to $SqlInstance"
+                    $CacheDP.Add($SqlInstance, '?')
+                    return '?'
+                }
+            }
+            if($CacheDP[$SqlInstance] -eq '?') {
+                return '?'
+            }
+            if(!$ComputerName) {
+                $ComputerName = $CacheDP[$SqlInstance].ComputerName
+            }
+            if(!$CacheMP[$ComputerName]) {
+                try {
+                    $CacheMP.Add($ComputerName, (Get-DbaDiskSpace -ComputerName $ComputerName -Credential $Credential))
+                } catch {
+                    Write-Warning "Can't connect to $ComputerName."
+                    $CacheMP.Add($ComputerName,'?')
+                    return '?'
+                }
+            }
+            if($DefaultPathType -eq 'Log') {
+                $Path = $CacheDP[$SqlInstance].Log
+            } else {
+                $Path = $CacheDP[$SqlInstance].Data
+            }
+            foreach($M in $CacheMP[$ComputerName]) {
+                if($Path -like "$($M.Name)*") {
+                    return $M.Name
+                }
+            }
+        }
+    }
+    process {
+        if(!$DestinationDatabase) {
+            $DestinationDatabase = $SourceDatabase
+        }
+        Write-Verbose "$Source.[$SourceDatabase] -> $Destination.[$DestinationDatabase]"
```

Comment:
This can be `Write-Message -Level VeryVerbose -Message "$($Source.[$SourceDatabase]) -> $($Destination.[$DestinationDatabase])"`